src/audio-manager.ts

```diff
@@ -0,0 +1,248 @@
+    private _destroy: boolean = false;
+
+    /**
+     * Constructs a PlayableNode.
+     *
+     * @warning This is for internal use only. PlayableNode's should only be created via the AudioManager's `load()`
+     * function.
+     * @param src Path to the audio file.
+     * @param audioBuffer Buffer of the decoded source.
+     * @param audioManager Manager that created the associated AudioBuffer.
+     */
+    constructor(src: string, audioBuffer: AudioBuffer, audioManager: AudioManager) {
+        this._audioBuffer = audioBuffer;
+        this._audioManager = audioManager;
+        this._source = src;
+        this._gainNode.connect(_audioContext.destination);
+    }
+
+    /**
+     * Asynchronously plays the loaded audio. If the audio is already playing, it stops the current playback and
+     * starts from the beginning.
+     * If the audio context is in a suspended state, it attempts to unlock the audio context before playing and will
+     * continue after the user has interacted with the website.
+     *
+     * @param posVec - An optional parameter representing the 3D spatial position of the audio source.
+     *                 If provided, the audio will be spatialized using a PannerNode based on the given position vector.
+     * @returns A Promise that resolves once the audio playback starts.
+     * @throws If there is an error during the playback process, a warning is logged to the console.
+     */
+    async play(posVec?: Float32Array): Promise<void> {
+        try {
+            if (this.isPlaying) {
+                this.stop();
+            }
+            if (_audioContext.state === 'suspended') {
+                await _unlockAudioContext();
+            }
+            this._audioNode = new AudioBufferSourceNode(_audioContext, {
+                buffer: this._audioBuffer,
+                loop: this.loop,
+            });
+            if (posVec !== undefined) {
+                this._pannerNode = new PannerNode(_audioContext, {
+                    coneInnerAngle: 360,
+                    coneOuterAngle: 0,
+                    coneOuterGain: 0,
+                    distanceModel: 'exponential' as DistanceModelType,
+                    maxDistance: 10000,
+                    refDistance: 1.0,
+                    rolloffFactor: 1.0,
+                    panningModel: this.HRTF ? 'HRTF' : 'equalpower',
+                    positionX: posVec[0],
+                    positionY: posVec[2],
+                    positionZ: -posVec[1],
+                    orientationX: 0,
+                    orientationY: 0,
+                    orientationZ: 1,
+                });
+                this._audioNode.connect(this._pannerNode).connect(this._gainNode);
+            } else {
+                this._audioNode.connect(this._gainNode);
+            }
+            this._audioNode.addEventListener('ended', this.stop);
+            this._audioNode.start();
+            this._isPlaying = true;
+        } catch (e) {
```

Comment:
Which part can throw exactly in this entire thing? Wondering if the try catch scope can be shorter or not